WordPress/gutenberg
# DataViews: do not display element descriptions in filters

Follow-up to https://github.com/WordPress/gutenberg/pull/64398#issuecomment-2301470485

## What?

Remove the description from the field elements.

| Before | After |
| --- | --- |
| <img width="315" alt="Captura de ecrã 2024-08-21, às 16 35 00" src="https://github.com/user-attachments/assets/07362617-8445-4d3a-899b-8c038c9ff837"> | <img width="316" alt="Captura de ecrã 2024-08-21, às 16 36 09" src="https://github.com/user-attachments/assets/90a52ea0-f992-49f8-9fc0-970daeaf3892"> |

## Why?

The elements in the fields are used for both filters and edit. In the edit form, some fields (status) wants to display descripionts, but this affected filters.

After proposing and discarding an API to have different elements in filters and edit context ([see](https://github.com/WordPress/gutenberg/pull/64674#discussion_r1724782006)), the way forward for now is not to display descriptions in filters.

## How?

Remove the existing display of descriptions in filters. This only affects the "Sync" filter in the Patterns page.

## Testing Instructions

Visit the Pages page in the site editor with the "quick edit" form experiment enabled:

- Verify that the elements for quick edit & the filter are different.
  - Edit should not contain `trash` and the descriptions are displayed.
  - Filter should contain `trash` but with no descriptions.

Visit the Patterns page in the site editor and verify that the filter doesn't contain descriptions.
